### PR TITLE
chore: refactor workloads

### PR DESCRIPTION
**Pull request by iliapolo**
The current class hierarchy around workloads (i.e `Job`, `Deployment`, `StatefulSet`, `Pod`) is fairly confusing.
Lots of name collisions and un-necessary interfaces. 

Also, right now, any new functionality that is added to a `Pod`, needs to be redefined for every workload that manages pods. 
This is a lot of boiler-plate code that can also cause oversights and bugs. 

> For example: https://github.com/cdk8s-team/cdk8s-plus/pull/469.

This PR remodels things a bit so that all common functionality is only implemented and defined inside the `Pod` class (sort of). It also introduces a new abstract class called `Workload`, that represents these [resources](https://kubernetes.io/docs/concepts/workloads/controllers/), and puts some common API into that as well.

Also, the `PodTemplate` helper class and all its derivatives were completely removed, I think it caused more confusion than benefit, it also collided with a possibly future `PodTemplate` resource.

### CODE REVIEW

File: src/pod.ts [hunk 1]
```diff
@@ -1,95 +1,12 @@
...
-   * Provides read/write access to the underlying pod metadata of the resource.
-   */
-  readonly podMetadata: ApiObjectMetadataDefinition;
-}
-
-/**
- * Provides read/write capabilities ontop of a `PodSpecProps`.
- */
-export class PodSpec implements IPodSpec {
+export abstract class AbstractPod extends Resource {
```
Comment (Chriscbr) [inline src/pod.ts]: Docstring?

**Review by Chriscbr — APPROVED** [review]
Love the simplification 👍

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-21/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/646)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-20/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/647)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**